docs/building.rst

```diff
@@ -647,3 +654,18 @@ in a command shell::
 
 This will also install the headers and library files in the visual studio directories to
 be picked up by libtorrent.
+
+list of macros
+--------------------
+
+The following is a list of defines that libtorrent is built with: ``BOOST_ALL_NO_LIB``, ``BOOST_ASIO_ENABLE_CANCELIO``, ``BOOST_ASIO_HAS_STD_CHRONO``, ``BOOST_MULTI_INDEX_DISABLE_SERIALIZATION``, ``BOOST_NO_DEPRECATED``, ``BOOST_SYSTEM_NO_DEPRECATED``, ``TORRENT_BUILDING_LIBRARY``, ``TORRENT_USE_I2P=1``
```

Comment:
are you sure that `TORRENT_USE_I2P` affects the ABI? if so, would you mind pointing out where? I thought I had fixed all the `TORREN_*` macros to not affect ABI (except for `TORRENT_ABI_VERSION`).

`TORRENT_BUILDING_LIBRARY` is *not* supposed to be defined by clients. It's there to allow libtorrent headers to tell when they are built as part of libtorrent itself or when they are included. On windows this is an important distinction since it's the difference between `__declspec(dllimpoty)` and `__declspec(dllexport)`

Reply:
TORRENT_USE_I2P - yeah, I wasn't too sure about it either. Ok, let's remove it.

TORRENT_BUILDING_LIBRARY - yes, makes sense, let's remove it.